dist/upload_maven.py

```diff
@@ -0,0 +1,137 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import argparse
+import base64
+import os
+import shutil
+import time
+
+try:
+    from urllib.request import urlopen, Request, HTTPError
+except ImportError:  # python 2
+    from urllib2 import urlopen, Request, HTTPError
+
+_USER_CREDS = os.environ.get("READWRITE_USER", "")
+_KEY_CREDS = os.environ.get("READWRITE_API_KEY", "")
+_ARTIFACT_HOST_URL = os.environ["ARTIFACT_HOST_URL"]
+
+_GROUP_ID = "io.envoyproxy.envoymobile"
+_ARTIFACT_ID = "envoy"
+_BASE_URL = "{}/io/envoyproxy/envoymobile".format(_ARTIFACT_HOST_URL)
+_LOCAL_INSTALL_PATH = os.path.expanduser("~/.m2/repository/io/envoyproxy/envoymobile/{}".format(_ARTIFACT_ID))
+
+
+def _upload(target, version):
+    aar = target + ".aar"
+    source_jar = target + "-sources.jar"  # TODO: figure out how to generate sources jar within bazel
+    pom_file = target + "-pom.xml"
+
+    files = [
+        (aar, "{}-{}.aar".format(_ARTIFACT_ID, version)),
+        (source_jar, "{}-{}-sources.jar".format(_ARTIFACT_ID, version)),
+        (pom_file, "{}-{}.pom".format(_ARTIFACT_ID, version))
+    ]
+
+    if version == "LOCAL-SNAPSHOT":
+        _install_locally(version, files)
+    else:
+        _upload_to_artifactory(version, files)
+
+
+def _upload_to_artifactory(version, files):
+    aar = files[0][0]
+    artifact_name = os.path.splitext(os.path.basename(aar))[0]
+
+    package_url = os.path.join(_BASE_URL, artifact_name, version)
+    base64string = base64.b64encode("{}:{}".format(_USER_CREDS,
+                                                   _KEY_CREDS)
+                                    .encode()).decode()
+
+    for src_path, target_path in files:
+        # Temporarily allow missing source jars
+        if not os.path.exists(src_path):
+            continue
+        basename = os.path.basename(target_path)
+        artifact_url = os.path.join(package_url, basename)
+        try:
+            with open(src_path, "rb") as f:
+                request = Request(artifact_url, f.read())
+
+            request.add_header("Authorization", "Basic {}".format(base64string))
+            request.get_method = lambda: "PUT"
+            urlopen(request)
+        except HTTPError as e:
+            if e.code == 403:
+                print("Ignoring duplicate upload for {}".format(artifact_url))
+            else:
+                print("Exception raised for {}".format(src_path))
+                print(e.headers)
+                raise
+        except Exception as e:
+            print(e)
+            raise
+
+
+def _install_locally(version, files):
```

Comment:
Do you still want this codepath?

Reply:
Yeah, I wanted to keep it so I can test things locally